### DISCUSSION TITLE
In the future, will it be possible to support both index of the partition and the collection

### DISCUSSION THREAD

**Discussion by shushenglee** [milvus-io/milvus#4878] [Ideas & Feature requests]
During the development of the project, in some cases we only need to retrieve in the partition, in other cases we need to retrieve in the collection, when I use the IVF_FLAT method to store the index, the retrieval speed in the collection of about 1kw data will be slow, sometimes it will be 10s-25s, but only 40ms-125ms when searching in the partition (containing 400w data).

I am wondering whether in a future version, it is possible to support a feature, that is, **the partition index and the collection index are independent of each other**.

**Comment by shengjun1985** [comment]
>A collection contains 1000w data and it costs 10s-25s.
>A partition contains 400w data and it costs 40ms-125ms.

It is pretty strange.
And I want to know why the cost doesn't grow linearly.
(1) Were the test commands the same, including "nq" and "topk"?
(2) Was your memory large enough and did all data store in the cache?

**Reply by shushenglee** [reply]
I am also very confused. 
(1) The test commands are the same, nq=1, topk=10;
(2) I would like to ask, how to judge that all data is in the cache? 
The following figure shows the memory usage of the milvus server:
![image](https://user-images.githubusercontent.com/25769282/113105070-c588db00-9233-11eb-896f-c432bc316626.png)
PS: cache parameters in server_config.yaml :
![image](https://user-images.githubusercontent.com/25769282/113105967-c79f6980-9234-11eb-81d8-103a2ef65334.png)

**Reply by shengjun1985** [reply]
I suspect your data is out of cache.
But the details need to be confirmed by log.

However, here is a tool which can estimate the memory size.
https://zilliz.com/sizing-tool

**Reply by shushenglee** [reply]
Thanks for providing the link. Maybe my data need 8GB cache or more.

**Comment by gujun720** [comment]
@shushenglee 
There is no global index on the collection level in Milvus.
The collection index actually consists of a series of partition indexes.

**Reply by shushenglee** [reply]
Oh, I see. Maybe I should locate the problem of slow query speed in the collection.

**Reply by tianya092** [reply]
感觉你的需求是，有时候需要在分区中查询，有时候需要在整个集合中查询。跟我一样，分区数过多后，整个集合查询会导致查询性能下降明显，每次需要4S，无法满足实时查询的场景

**Comment by tianya092** [comment]
有时候需要在分区中查询，有时候需要在整个集合中查询。跟我一样，分区数过多后，整个集合查询会导致查询性能下降明显，每次需要4S，无法满足实时查询的场景
